src/smt/smt_engine.cpp

```diff
@@ -1925,10 +1927,13 @@ void SmtEngine::printStatisticsSafe(int fd) const
   d_env->getStatisticsRegistry().printSafe(fd);
 }
 
-void SmtEngine::printStatisticsDiff(std::ostream& out) const
+void SmtEngine::printStatisticsDiff() const
 {
-  d_env->getStatisticsRegistry().printDiff(out);
-  d_env->getStatisticsRegistry().storeSnapshot();
+  if (d_env->getOptions().base.statisticsEveryQuery)
```

Comment:
Maybe move this to guard the call to this method, not guard within the method itself.